integr8ly/cloud-resource-operator
# MGDAPI-4900 [Postgres] Update

## Overview
Jira: [MGDAPI-4900](https://issues.redhat.com/browse/MGDAPI-4900)


## Verification
- Provision a GCP cluster
    - From the master branch of the [Delorean](https://github.com/integr8ly/delorean) repo create a file in the root of the repo called gcp_service_account.json. Request GCP credentials and add these to this file.
    - Run `make -f make/ocm.mk ocm/cluster.json BYOC=true CLOUD_PROVIDER=gcp`
    - Edit the cluster.json file to update the name (and remove the expiration timestamp if necessary) You can also reduce the number of nodes to 3.
    - Run `make ocm/cluster/create`

- Create a Postgres instance
   - Checkout this pull request `gh pr checkout <PR>`
   - Log in to your cluster and run `PROVIDER=gcp make cluster/prepare`
   - Run cro `make run`
   - Create a Postgres CR `PROVIDER=gcp make cluster/seed/postgres` and wait for it to provision.
   - On the command line view the current list of instances `gcloud sql instances list`
   - Alternatively you can view your instance on the GCP [console](https://console.cloud.google.com/sql/instances?project=rhoam-317914).

- Update the Postgres instance by patching the strategy config map, and setting the maintenance window to true.
   - Update the database version :
   ``` 
   oc patch configmap cloud-resources-gcp-strategies -n cloud-resource-operator --type=json -p'[{"op":"replace","path":"/data/postgres","value":"{\"development\":{\"region\":\"\",\"projectID\":\"\",\"createStrategy\":{\"instance\":{\"databaseVersion\":\"POSTGRES_14\"}},\"deleteStrategy\":{}}}"}]' 
   oc patch postgres example-postgres -n cloud-resource-operator --type=json -p='[{"op":"replace","path":"/spec/maintenanceWindow","value":true}]'
   ```
   - Wait until it updates and check that it has updated correctly by viewing the instance details
 `gcloud sql instances describe <INSTANCE_NAME>`

   - Update another setting (Other settings cannot be changed at the same time as the database version).
   ```
   oc patch configmap cloud-resources-gcp-strategies -n cloud-resource-operator --type=json -p'[{"op":"replace","path":"/data/postgres","value":"{\"development\":{\"region\":\"\",\"projectID\":\"\",\"createStrategy\":{\"instance\":{\"settings\":{\"storageAutoResizeLimit\":\"80\"}}},\"deleteStrategy\":{}}}"}]' 
   oc patch postgres example-postgres -n cloud-resource-operator --type=json -p='[{"op":"replace","path":"/spec/maintenanceWindow","value":true}]'
   ```

    - Wait until it updates and check that it has updated correctly by viewing the instance details
    `gcloud sql instances describe <INSTANCE_NAME>`

 - Delete the Postgres CR and GCP Postgres Instance
 `oc delete postgres example-postgres -n cloud-resource-operator`
